pkg/karmadactl/deinit.go

```diff
@@ -0,0 +1,345 @@
+	karmadaNodeLabel := "karmada.io/etcd"
+	nodeClient := d.KubeClientSet.CoreV1().Nodes()
+	nodes, err := nodeClient.List(context.TODO(), metav1.ListOptions{LabelSelector: karmadaNodeLabel})
+	if err != nil {
+		return err
+	}
+	if len(nodes.Items) == 0 {
+		fmt.Printf("node not found by label %q\n", karmadaNodeLabel)
+		return nil
+	}
+
+	for v := range nodes.Items {
+		removeLabels(&nodes.Items[v], karmadaNodeLabel)
+		fmt.Printf("remove node %q labels %q\n", nodes.Items[v].Name, karmadaNodeLabel)
+		if _, err := nodeClient.Update(context.TODO(), &nodes.Items[v], metav1.UpdateOptions{}); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func removeLabels(node *corev1.Node, removesLabel string) {
+	for label := range node.Labels {
+		if strings.Contains(label, removesLabel) {
+			delete(node.Labels, label)
+		}
+	}
+}
+
+// RenameKarmadaDataPath backup karmada directory
+func (d *CommandDeInitOption) RenameKarmadaDataPath() error {
+	if !isExist(d.KarmadaDataPath) {
+		fmt.Printf("WARNING %q file does not exist.\n", d.KarmadaDataPath)
+		return nil
+	}
+	timeStr := time.Now().Format("20060102150405")
+	backupName := fmt.Sprintf("%s-%s", d.KarmadaDataPath, timeStr)
+	if err := os.Rename(d.KarmadaDataPath, backupName); err != nil {
+		return err
+	}
+	return nil
+}
+
+// DeleteNamespace Delete namespace
+// TODO k8s will recursively delete all resources under namespace,including etcd persistent storage.
+func (d *CommandDeInitOption) DeleteNamespace() error {
+	fmt.Printf("Delete namespace %q\n", d.Namespace)
+	if err := d.KubeClientSet.CoreV1().Namespaces().Delete(context.TODO(), d.Namespace, metav1.DeleteOptions{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// deleteConfirmation delete confirmation karmada
+func deleteConfirmation() bool {
+	fmt.Println("Please type (y)es or (n)o and then press enter:")
+	var response string
+	_, err := fmt.Scanln(&response)
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	switch strings.ToLower(response) {
+	case "y", "yes":
+		return true
+	case "n", "no":
+		return false
+	default:
+		return deleteConfirmation()
+	}
+}
+
+// RunDeInit start delete action
+func (d *CommandDeInitOption) RunDeInit() error {
+	// delete confirmation,exit the delete action when false.
+	fmt.Println("remove karmada on kubernetes, please make data backup.")
+	if !d.Force {
+		d.Force = deleteConfirmation()
+	}
+	if !d.Force {
+		os.Exit(0)
+	}
+
+	if err := d.DeleteDeployments(); err != nil {
+		return err
+	}
+
+	if err := d.DeleteStatefulSets(); err != nil {
+		return err
+	}
+
+	if err := d.DeleteServices(); err != nil {
+		return err
+	}
+
+	if err := d.DeleteSecrets(); err != nil {
+		return err
+	}
+
+	if err := d.DeleteClusterRole(); err != nil {
+		return err
+	}
+
+	if err := d.RemoveNodeLabels(); err != nil {
+		return err
+	}
+
+	if err := d.RenameKarmadaDataPath(); err != nil {
+		return err
+	}
+
+	if d.Recursive {
+		if err := d.DeleteNamespace(); err != nil {
+			return err
+		}
+	}
```

Comment:
If we don't remove `kube-system` namespace by default, would it block installing Karmada again?

Reply:
Will not